scripts/run_social_messaging.py

```diff
@@ -0,0 +1,167 @@
+            sockets = dict(poller.poll(500))  # 500 ms timeout
+
+            if cmd_social_subscriber.socket in sockets and \
+                    sockets[cmd_social_subscriber.socket] == zmq.POLLIN:
+
+                msg_type, msg_obj = cmd_social_subscriber.receive_message(flags=zmq.NOBLOCK)
+
+                # Check the various social sinks
+                if social_twitter is not None:
+                    social_twitter.send_message(msg_obj['message'], msg_obj['timestamp'])
+
+                if social_slack is not None:
+                    social_slack.send_message(msg_obj['message'], msg_obj['timestamp'])
+
+            time.sleep(1)
+    except KeyboardInterrupt:
+        pass
+
+
+def run_social_sinks(msg_port, social_twitter, social_slack):
+    the_root_logger.info('Creating sockets')
+
+    threads = []
+    
+    name='social_messaging'
+
+    t = threading.Thread(
+            target=check_social_messages_loop, name=name, args=(msg_port, social_twitter, social_slack), daemon=True)
+    the_root_logger.info('Starting thread {}', name)
+    t.start()
+    threads.append(t)
+
+    time.sleep(0.05)
+    if not any([t.is_alive() for t in threads]):
+        say('Failed to start social sinks thread!', error=True)
+        sys.exit(1)
+    else:
+        the_root_logger.info('Started social messaging')
+        print()
+        print('Hit Ctrl-c to stop')
+    try:
+        # Keep running until they've all died.
+        while threads:
+            for t in threads:
+                t.join(timeout=100)
+                if t.is_alive():
+                    continue
+                say('Thread {} has stopped', t.name, error=True)
+                threads.remove(t)
+                break
+        # If we get here, then the forwarders died for some reason.
+        sys.exit(1)
+    except KeyboardInterrupt:
+        sys.exit(0)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description='Run social messaging to forward platform messages to social channels.')
+    parser.add_argument(
+        '--from_config',
+        action='store_true',
+        help='Read social channels config from the pocs.yaml and pocs_local.yaml config files.')
+    args = parser.parse_args()
+
+    def arg_error(msg):
+        print(msg, file=sys.stderr)
+        parser.print_help()
+        sys.exit(1)
+
+    # Initialise social sinks to None
+    social_twitter = None
+    social_slack = None
+    
+    if args.from_config:
+        config = load_config(config_files=['pocs'])
+        
+        if 'social_accounts' in config:
+            social_config = config['social_accounts']
+
+            # See which social sink we can create based on config
+
+            # Twitter sink
+            if social_config is not None and 'twitter' in social_config:
+                twitter_config = social_config['twitter']
+                if twitter_config is not None and \
+                        'consumer_key' in twitter_config and \
```

Comment:
If you got with my suggestion earlier, this validation will move into the twitter class. And you could of course use a factory method rather than the constructor to do your validation, if that made things cleaner.

Reply:
See my comment above about handling config in main module